generator/internal/stub_generator.cc

```diff
@@ -88,16 +85,37 @@ Status StubGenerator::GenerateHeader() {
   header_.Print("\n");
 
   // Abstract interface Stub base class
-  header_.Print(vars_,  // clang-format off
+  header_.Print(service_vars_,  // clang-format off
     "class $stub_class_name$ {\n"
     " public:\n"
     "  virtual ~$stub_class_name$() = 0;\n"
     "\n");
   // clang-format on
 
+  for (int i = 0; i < service_descriptor_->method_count(); i++) {
+    auto method = service_descriptor_->method(i);
+    std::map<std::string, std::string> vars(service_vars_.begin(),
+                                            service_vars_.end());
+    vars.insert(service_method_vars_[method->full_name()].begin(),
```

Comment:
I've squinted at this line for a couple minutes, and I'm sure whether it's safe in the case where the full_name is not found in the map and `op[]` needs to insert a new element. To avoid the question, you could pull the `op[]` call out into a separate statement, like:

```cc
auto const method_vars = service_method_vars_[method->full_name()];
vars.insert(method_vars.begin(), method_vars.end());
```